File: backend/src/main/kotlin/ch/streckeisen/mycv/backend/publicapi/PublicProfileDto.kt
package ch.streckeisen.mycv.backend.publicapi

import java.time.LocalDate

data class PublicProfileDto(
    val id: Long?,
    val firstName: String,
    val lastName: String,
    val email: String,
    val phone: String,
    val birthday: LocalDate,
    val street: String,
    val houseNumber: String?,
    val postcode: String,
    val city: String,
    val country: String,
    val workExperiences: List<PublicWorkExperienceDto>,
    val skills: List<PublicSkillDto>,
    val education: List<PublicEducationDto>,
)

File: frontend/src/dto/PublicProfileDto.ts
import type { PublicWorkExperienceDto } from '@/dto/PublicWorkExperienceDto'
import type { PublicSkillDto } from '@/dto/PublicSkillDto'
import type { PublicEducationDto } from '@/dto/PublicEducationDto'
import type { PublicProfileThemeDto } from '@/dto/PublicProfileThemeDto'
import { PublicProjectDto } from '@/dto/PublicProjectDto'

export type PublicProfileDto = {
  firstName: string
  lastName: string
  jobTitle: string
  bio: string
  email?: string
  phone?: string
  address?: PublicAddressDto
  profilePicture: string
  workExperiences: Array<PublicWorkExperienceDto>
  skills: Array<PublicSkillDto>
  education: Array<PublicEducationDto>
  projects: Array<PublicProjectDto>
  theme?: PublicProfileThemeDto
  language: string
}

export type PublicAddressDto = {
  street: string
  houseNumber?: string
  postcode: string
  city: string
  country: string
}
